packages/flutter/lib/src/material/ink_well.dart

```diff
@@ -1104,6 +1104,7 @@ class _InkResponseState extends State<_InkResponseStateWidget>
               onLongPress: widget.onLongPress != null ? () => _handleLongPress(context) : null,
               behavior: HitTestBehavior.opaque,
               excludeFromSemantics: widget.excludeFromSemantics,
+              excludeTapFromSemantics: widget.onTap == null,
```

Comment:
Why do we provide a tap handler to onTap in the first place if widget.onTap is null?

Reply:
Seems like it is a part of the inkwell logic - it tracks tap down/up/etc for the splash reactions - I tried just removing the onTap altogether if one was not provided and it left the splashes on the button.

Maybe @HansMuller might have some better idea of the history here?

Reply:
Can we specify a separate onTapUp to remove the splashes added by onTapDown, but only provide onTap if there's an actual onTap handler provided?

Reply:
I have no idea what the InkWell should do

Reply:
(it looks like down starts the splash, and onTap "confirms" it?)

Reply:
Moving the confirmation to onTapUp would keep the current behavior since onTapUp is called just before onTap (and only if onTap is called). However, you're right, I am not sure if InkWell is supposed to behave like this. If you don't have an onTap handler (just say a longPress Handler) and an InkWell is single-Tapped, should that really show the "splash confirmed" animations as it is doing now? Seems odd... Let's check with a material expert :)

Reply:
@goderbauer can you escalate this issue? It is tagged as a P1, we need someone to look at it.